CMakeLists.txt

```diff
@@ -7,39 +8,80 @@ include($ENV{ROS_ROOT}/core/rosbuild/rosbuild.cmake)
 #  Release        : w/o debug symbols, w/ optimization
 #  RelWithDebInfo : w/ debug symbols, w/ optimization
 #  MinSizeRel     : w/o debug symbols, w/ optimization, stripped binaries
-#set(ROS_BUILD_TYPE RelWithDebInfo)
 
-rosbuild_init()
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake/Modules/)
 
 find_package(Jsoncpp REQUIRED)
+find_package(catkin REQUIRED COMPONENTS
+  rospy
+  roscpp
+  std_msgs
+  geometry_msgs
+  cv_bridge
+  image_transport
+  camera_info_manager
+  message_generation
+)
+
+add_message_files(FILES
+    Activities.msg
+    Appearance.msg
+    Audio.msg    
+    BodyArray.msg
+    Expressions.msg
+    JointOrientationAndType.msg
+    JointPositionAndState.msg
+    Lean.msg
+    Body.msg
+)
+
+#uncomment if you have defined messages/services
+generate_messages(DEPENDENCIES geometry_msgs)
+
+
+catkin_package(
+    INCLUDE_DIRS include
+    #LIBRARIES
+    #    ${PROJECT_NAME}
+    CATKIN_DEPENDS
+        geometry_msgs
+        std_msgs
+    DEPENDS
+        Jsoncpp
+)
 
 #set the default path for built executables to the "bin" directory
 set(EXECUTABLE_OUTPUT_PATH ${PROJECT_SOURCE_DIR}/bin)
 #set the default path for built libraries to the "lib" directory
 set(LIBRARY_OUTPUT_PATH ${PROJECT_SOURCE_DIR}/lib)
 
-#uncomment if you have defined messages
-rosbuild_genmsg()
-#uncomment if you have defined services
-#rosbuild_gensrv()
-
 set(includeFolders ${PROJECT_SOURCE_DIR}/include/k2_client)
```

Comment:
Can we do
`set(${PROJECT_NAME}_INCLUDE_DIRS ${PROJECT_SOURCE_DIR}/include/k2_client)`?